### PR TITLE
Add some more functionality, docs and tests for root systems

### PR DESCRIPTION
**Pull request by lgoettgens**

### CODE REVIEW

**Comment by felix-roehrich** [comment]
Please add a property test to make sure that the roots of the dual root system correspond to the coroots of the base root system.

File: experimental/LieAlgebras/src/RootSystem.jl [hunk 1]
```diff
@@ -81,11 +89,41 @@ function cartan_matrix(R::RootSystem)
...
+  return D
+end
+
+@doc raw"""
+    coroot(R::RootSystem, i::Int) -> RootSpaceElem
+
+Returns the `i`-th coroot of `R`, i.e. the `i`-th root of the dual root system of `R`.
+This is a more efficient version for `coroots(R)[i]`.
+
+Also see: `coroots`.
```
Comment (fingolfin) [inline experimental/LieAlgebras/src/RootSystem.jl:100]: How about using an actual ref?
```suggestion
Also see: [`coroots`](@ref).
```

Comment (lgoettgens) [reply experimental/LieAlgebras/src/RootSystem.jl:100]: This does only work once we have a markdown page collecting the docstrings. That's something @felix-roehrich wants to approach in the near future.

**Comment by lgoettgens** [comment]
> Please add a property test to make sure that the roots of the dual root system correspond to the coroots of the base root system.

done

File: experimental/LieAlgebras/src/RootSystem.jl [hunk 2]
```diff
@@ -72,6 +78,10 @@ function Base.show(io::IO, R::RootSystem)
   end
 end
 
+@attr ZZMatrix function cartan_bilinear_form(R::RootSystem)
```
Comment (felix-roehrich) [inline experimental/LieAlgebras/src/RootSystem.jl]: I think we agreed that here `cartan` is not necessary.

```suggestion
@attr ZZMatrix function bilinear_form(R::RootSystem)
```

File: experimental/LieAlgebras/src/RootSystem.jl [hunk 3]
```diff
@@ -81,11 +91,34 @@ function cartan_matrix(R::RootSystem)
...
+    coroots(R::RootSystem) -> Vector{RootSpaceElem}
+
+Returns the coroots of `R`, starting with the coroots of positive roots and then the negative roots,
+in the order of `positive_coroots` and `negative_coroots`.
+
+Also see: `coroot`.
+"""
+function coroots(R::RootSystem)
+  return [[r for r in positive_coroots(R)]; [-r for r in positive_coroots(R)]]
+  (dual(R))
```
Comment (felix-roehrich) [inline experimental/LieAlgebras/src/RootSystem.jl]:
```suggestion
```

File: experimental/LieAlgebras/src/RootSystem.jl [hunk 4]
```diff
@@ -313,6 +514,10 @@ function Base.iszero(r::RootSpaceElem)
   return iszero(r.vec)
 end
 
+function Base.length(r::RootSpaceElem)
+  return dot(r, r)
```
Comment (felix-roehrich) [inline experimental/LieAlgebras/src/RootSystem.jl]: We need `sqrt` here, but does this work with current types ?
```suggestion
  return sqrt(dot(r, r))
```

Comment (lgoettgens) [reply experimental/LieAlgebras/src/RootSystem.jl]: No, that doesn't work

Comment (lgoettgens) [reply experimental/LieAlgebras/src/RootSystem.jl]: We could instead name the function something like `length_squared` instead

Comment (felix-roehrich) [reply experimental/LieAlgebras/src/RootSystem.jl]: That is neither intuitive nor shorter than `dot(r, r)`, so I don't feel great about this.

Comment (lgoettgens) [reply experimental/LieAlgebras/src/RootSystem.jl]: The let's just remove `length`

File: experimental/LieAlgebras/src/LieAlgebras.jl [hunk 5]
```diff
@@ -73,9 +75,10 @@ import ..Oscar:
   ⊕,
   ⊗
 
-import Base: getindex, deepcopy_internal, hash, issubset, iszero, parent, zero
+import Base: getindex, deepcopy_internal, hash, issubset, iszero, length, parent, zero
```
Comment (lgoettgens) [inline experimental/LieAlgebras/src/LieAlgebras.jl]:
```suggestion
import Base: getindex, deepcopy_internal, hash, issubset, iszero, parent, zero
```

File: experimental/LieAlgebras/src/RootSystem.jl [hunk 6]
```diff
@@ -313,6 +513,10 @@ function Base.iszero(r::RootSpaceElem)
...
+end
+
```
Comment (lgoettgens) [inline experimental/LieAlgebras/src/RootSystem.jl]:
```suggestion
```

**Review by felix-roehrich — APPROVED** [review]

**Comment by lgoettgens** [comment]
can we get a merge of this? maybe @thofma ?

**Review by thofma — APPROVED** [review]

**Pull request merged**